### PR TITLE
net: dhcpv4_server: Several improvements for issues found while testing

### PR DESCRIPTION
**Pull request by rlubos**
During field tests of DHCPv4 server running on Wi-FI AP, a few improvements were needed for robust operation.
Please check the commit messages for details.

### CODE REVIEW

**Comment by rlubos** [comment]
CC @simensrostad

File: subsys/net/lib/dhcpv4/Kconfig [hunk 1]
```diff
@@ -148,4 +148,17 @@ config NET_DHCPV4_SERVER_ICMP_PROBE_TIMEOUT
 	  request) and wait for the time specific by this config before offering
 	  the address. If set to 0, ICMP probing will be disabled.
 
+config NET_DHCPV4_SERVER_NAK_UNRECOGNIZED_REQUESTS
+	bool "Reply with NAK for DHCP Requests from clients we do not recognize"
+	default y
+	help
+	  Zephyr's DHCPv4 server does not implement persistent storage of address
+	  leases. In result, all leases are lost on reboot, which can cause
```
Comment (pdgendt) [inline subsys/net/lib/dhcpv4/Kconfig:156]:
```suggestion
	  leases. As a result, all leases are lost on reboot, which can cause
```

File: subsys/net/lib/dhcpv4/Kconfig [hunk 2]
```diff
@@ -148,4 +148,17 @@ config NET_DHCPV4_SERVER_ICMP_PROBE_TIMEOUT
...
+	  delays with clients starting in INIT-REBOOT state and thus sending
+	  (potentially several) Requests before attempting full Discover-Request
+	  procedure. Enabling this config will violate the RFC in this regard, as
+	  instead of staying silent for unrecognized Request messages, the server
+	  will send NAK, informing the client it should proceed with full
```
Comment (pdgendt) [inline subsys/net/lib/dhcpv4/Kconfig:161]:
```suggestion
	  will send a NAK, informing the client it should proceed with a full
```

**Review by pdgendt — DISMISSED** [review]

**Comment by rlubos** [comment]
Thanks @pdgendt, I've applied your suggestions.

**Review by pdgendt — APPROVED** [review]

**Review by maass-hamburg — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by simensrostad — APPROVED** [review]

**Pull request merged**